code/modules/crew_objectives/civilian_objectives.dm

```diff
@@ -54,16 +54,50 @@
 	else
 		return ..()
 
-/datum/objective/crew/responsibility
-	explanation_text = "Make sure nobody dies with alcohol poisoning."
+/datum/objective/crew/cocktail
+	explanation_text = "Have a bottle(any type) that contains 'something' when the shift ends. Each of them must be at least 'something'u."
 	jobs = "bartender"
+	var/targetchems = list()
+	var/chemnames = ""
+	var/chemsize
+	var/datum/reagent/chempath
 
-/datum/objective/crew/responsibility/check_completion()
-	for(var/mob/living/carbon/human/H in GLOB.mob_list)
-		if(H.stat == DEAD && H.drunkenness >= 80)
-			if((H.z in SSmapping.levels_by_trait(ZTRAIT_STATION)) || SSshuttle.emergency.shuttle_areas[get_area(H)])
-				return ..()
-	return TRUE
+/datum/objective/crew/cocktail/New()
+	. = ..()
+	var/mydrink = list()
+	for(var/i in 1 to 5)
+		chempath = get_random_reagent_id(CHEMICAL_GOAL_BARTENDER_SERVING)
+		if(!(chempath in targetchems))
+			targetchems += chempath
+			mydrink += "[initial(chempath.name)]"
+	for(var/i in 1 to length(mydrink)-1)
+		chemnames += "[mydrink[i]], "
+	// chems may reaction, but there's no reactionable recipe from CHEMICAL_GOAL_BARTENDER_SERVING. Just don't put basic chems there.
+	chemnames += "and [mydrink[length(mydrink)]]"
+	qdel(mydrink)
+	chemsize = 4+(5-length(targetchems))
+	update_explanation_text()
+
+/datum/objective/crew/cocktail/update_explanation_text()
+	. = ..()
+	explanation_text = "Have a bottle(any type) that contains '[chemnames]' when the shift ends. Each of them must be at least [chemsize]u."
+
+/datum/objective/crew/cocktail/check_completion()
+	var/count = length(targetchems)
+	if(owner.current)
+		if(owner.current.contents)
+			// check every bottle in your bag.
+			for(var/obj/item/reagent_containers/B in owner.current.get_contents())
+				count = length(targetchems) // a bottle should have the all desired chems. reset the count for every try.
```

Comment:
This should be defined here

Reply:
count var exists at the parent.

Reply:
Move it up then, its not used outside of this local subblock so it doesn't need the full function scope

Reply:
No it exists on line 86

Reply:
![image](https://user-images.githubusercontent.com/26465327/179415498-f44dfb12-4eef-4804-8c60-1bf306873dac.png)

Reply:
ah, right. I was blind